### ISSUE TITLE
Configuration for numpy

### ISSUE THREAD

**Issue by shaunc** [microsoft/pylance-release#5307]
Pylance is not recognizing arithmetic using numpy types:

![image](https://github.com/microsoft/pylance-release/assets/193170/e732deb8-d756-4b37-b2da-993b46b80f1d)

I am using numpy 1.26.1, pylance v2023.12.1 and python 3.11.

I am using "strict" checking - I assume that that should be possible.

As others are not getting this error, I assume there is something strange about my configuration.  I use mypy in pre-commit to check these errors, and have mypy config in pyproject.toml:
```toml

[tool.mypy]
allow_untyped_calls = false
follow_imports = "skip"
ignore_missing_imports = true
plugins = "numpy.typing.mypy_plugin"
python_version = "3.11"
```

Does pylance require to be told about the (equivalent of?) the mypy_plugin for numpy, and if so, how do I configure it? 

Or perhaps the problem is a cache somewhere? This used to work, before stopping at some point.

**Comment by aCatasaurus** [comment]
Same issue. After closing and reopening VScode, Pylance suddenly does not recognize any standard operations on numpy ints.

![image](https://github.com/microsoft/pylance-release/assets/90944277/b7f1076f-7a46-45e7-92af-8ee2a4deaa30)

Environment:
 - Windows 10
 - Python 3.11.4
 - NumPy 1.25.2
 - Pylance v2023.12.1

The only setting I've configured for Pylance is the typechecking mode.
settings.json: `"python.analysis.typeCheckingMode": "basic"`

It was working fine just a few hours ago. The only change I can think of is opening another project before swapping back.

**Comment by debonte** [comment]
This is a duplicate of https://github.com/microsoft/pyright/issues/6466 which was fixed in Pyright 1.1.337. That fix is included in our latest prerelease builds.

You can eliminate the issue by switching to Pylance prerelease.
